tests/functional/test_fusion_hap.py

```diff
@@ -0,0 +1,1024 @@
+    m_hap_api.return_value = api_obj
+
+    # mock operation results
+    op_obj = MagicMock()
+    op_obj.get_operation = MagicMock(return_value=FailedOperationMock)
+    m_op_api.return_value = op_obj
+
+    # run module
+    with pytest.raises(exec_catch):
+        fusion_hap.main()
+
+    # check api was called correctly
+    api_obj.list_host_access_policies.assert_called_once_with()
+    api_obj.get_host_access_policy.assert_called_once_with(
+        host_access_policy_name=module_args["name"]
+    )
+    api_obj.create_host_access_policy.assert_not_called()
+    api_obj.update_host_access_policy.assert_not_called()
+    api_obj.delete_host_access_policy.assert_not_called()
+    op_obj.get_operation.assert_not_called()
+
+
+@patch("fusion.OperationsApi")
+@patch("fusion.HostAccessPoliciesApi")
+def test_hap_update(m_hap_api, m_op_api, module_args, current_hap_list):
+    # NOTE: Host Access Policy does not have PATCH method, thus no action is expected
+    current_hap = current_hap_list.items[0]
+    module_args["name"] = current_hap.name
+    module_args["display_name"] = "New Display Name"
+    module_args["iqn"] = current_hap.iqn
+    module_args["personality"] = (
+        "windows" if current_hap.personality != "windows" else "linux"
+    )
+    set_module_args(module_args)
+
+    # mock api responses
+    api_obj = MagicMock()
+    api_obj.list_host_access_policies = MagicMock(return_value=current_hap_list)
+    api_obj.get_host_access_policy = MagicMock(return_value=current_hap)
+    api_obj.create_host_access_policy = MagicMock(return_value=OperationMock(1))
+    api_obj.update_host_access_policy = MagicMock(return_value=OperationMock(2))
+    api_obj.delete_host_access_policy = MagicMock(return_value=OperationMock(3))
+    m_hap_api.return_value = api_obj
+
+    # mock operation results
+    op_obj = MagicMock()
+    op_obj.get_operation = MagicMock(return_value=SuccessfulOperationMock)
+    m_op_api.return_value = op_obj
+
+    # run module
+    with pytest.raises(AnsibleExitJson) as exc:
+        fusion_hap.main()
+
+    assert exc.value.args[0]["changed"] is False
+
+    # check api was called correctly
+    api_obj.list_host_access_policies.assert_called_once_with()
+    api_obj.get_host_access_policy.assert_called_once_with(
+        host_access_policy_name=module_args["name"]
+    )
+    api_obj.create_host_access_policy.assert_not_called()
+    api_obj.update_host_access_policy.assert_not_called()
+    api_obj.delete_host_access_policy.assert_not_called()
+    op_obj.get_operation.assert_not_called()
+
+
+@pytest.mark.skip(reason="Host Access Policy does not have PATCH method yet")
+@patch("fusion.OperationsApi")
+@patch("fusion.HostAccessPoliciesApi")
+@pytest.mark.parametrize(
+    "exec_original,exec_catch",
+    [
+        (purefusion.rest.ApiException, purefusion.rest.ApiException),
+        (HTTPError, HTTPError),
+    ],
+)
+def test_az_update_exception(
```

Comment:
These stubs should probably be named `test_hap_...`. However to tell the truth I would not add these stubs at all. By doing so you're creating an expectation which does not necessarily have to be met, since it's out of our control and can confuse the users.

Reply:
If you want to have a reminder, create tasks for it rather than leaving such stubs. Tasks are actionable.

Reply:
You're right. I will remove it also from the `test_fusion_az.py` file.